### PR TITLE
Update on biased MD (ABF) of ditopic molecules

### PR DESCRIPTION
**Pull request by franklhy**
(1) Add an ABF logger.
(2) Add a new collective variable: sum of two distances
(3) Update residue database

### CODE REVIEW

**Review by p-zubieta — APPROVED** [review]

File: silc/force_field/gaff2.py [hunk 1]
```diff
@@ -283,10 +283,12 @@ def set_dummy_replacement(self, dummy_replacement):
             raise RuntimeError("Please set dummy replacement first using set_dummy_replacement().")
         if isinstance(dummy_replacement, list) and len(dummy_replacement) == self.num_dummy:
             for i in range(len(dummy_replacement)):
-                if len(dummy_replacement[i]) == 1:
+                if len(dummy_replacement[i]) == 1 or dummy_replacement[i] == "[NH3+]":
                     self.replace_dummy_with.append(dummy_replacement[i])
```
Comment (p-zubieta) [inline silc/force_field/gaff2.py]: Not that important, but we don't really need the `i` here, right? If so, I think you can simplify this a bit, like so:
```suggestion
            for dummy in dummy_replacement:
                if len(dummy) == 1 or dummy == "[NH3+]":
                    self.replace_dummy_with.append(dummy)
```

Comment (franklhy) [reply silc/force_field/gaff2.py]: Sure, I will simplify it. Thanks for the suggestion.

**Comment by p-zubieta** [comment]
Not convinced about the names of the CVs, but otherwise looks good to me.

**Comment by franklhy** [comment]
> Not convinced about the names of the CVs, but otherwise looks good to me.

yeah, we will change the name later in a new PR.

**Pull request merged**